test_runner/performance/test_bulk_insert.py

```diff
@@ -28,3 +31,38 @@ def test_bulk_insert(neon_with_baseline: PgCompare):
 
             env.report_peak_memory_use()
             env.report_size()
+
+    # When testing neon, also check how long it takes the pageserver to reingest the
+    # wal from safekeepers. If this number is close to total runtime, then the pageserver
+    # is the bottleneck.
+    if isinstance(env, NeonCompare):
+        measure_recovery_time(env, test_output_dir)
+
+
+def measure_recovery_time(env: NeonCompare, test_output_dir):
+    client = env.env.pageserver.http_client()
+    pg_version = PgVersion(client.timeline_detail(env.tenant, env.timeline)["pg_version"])
+
+    def get_lsn():
+        return client.timeline_detail(env.tenant, env.timeline)["last_record_lsn"]
+
+    target_lsn = get_lsn()
+
+    # Stop pageserver and remove tenant data
+    env.env.pageserver.stop()
+    timeline_dir = (
+        test_output_dir / "repo" / "tenants" / str(env.tenant) / "timelines" / str(env.timeline)
+    )
```

Comment:
```suggestion
    timeline_dir = env.env.timeline_dir(env.tenant, env.timeline)
```

Which will also simplify the `test_output_dir` away as well..